pkg/workspace/controllers/workspace_gc_finalizer.go

```diff
@@ -29,6 +29,7 @@ func (c *WorkspaceReconciler) garbageCollectWorkspace(ctx context.Context, wObj
 	// We should delete all the nodeClaims that are created by this workspace
 	for i := range ncList.Items {
 		if ncList.Items[i].DeletionTimestamp.IsZero() {
+			klog.InfoS("Deleting associated NodeClaim...", "nodeClaim", ncList.Items[i].Name)
 			if deleteErr := c.Delete(ctx, &ncList.Items[i], &client.DeleteOptions{}); deleteErr != nil {
```

Comment:
I am considering adding a timeout here. Is this delete a synchronous call? Will it wait for claim and its node to be deleted? Or just marks for deletion? @helayoty @rambohe-ch

Reply:
The delete call is async by default. If we add the finalizer, it will wait until all related objects are deleted before deleting the nodeclaim.